> `cjson.override { stdenv = clangStdenv; }` already builds for me without this PR (x64 Linux).

What about `pkgsLLVM.cjson`? I get this with `pkgsLLVM`:
```
cjson> [  2%] Building C object CMakeFiles/cjson.dir/cJSON.c.o
cjson> [  4%] Building C object tests/CMakeFiles/unity.dir/unity/src/unity.c.o
cjson> /build/source/cJSON.c:127:5: error: 'sprintf' was marked unused but was used [-Werror,-Wused-but-marked-unused]
cjson>   127 |     sprintf(version, "%i.%i.%i", CJSON_VERSION_MAJOR, CJSON_VERSION_MINOR, CJSON_VERSION_PATCH);
cjson>       |     ^
cjson> /build/source/cJSON.c:571:18: error: 'sprintf' was marked unused but was used [-Werror,-Wused-but-marked-unused]
cjson>   571 |         length = sprintf((char*)number_buffer, "null");
cjson>       |                  ^
cjson> /build/source/cJSON.c:575:12: error: 'sprintf' was marked unused but was used [-Werror,-Wused-but-marked-unused]
cjson>   575 |                 length = sprintf((char*)number_buffer, "%d", item->valueint);
cjson>       |                          ^
cjson> /build/source/cJSON.c:580:18: error: 'sprintf' was marked unused but was used [-Werror,-Wused-but-marked-unused]
cjson>   580 |         length = sprintf((char*)number_buffer, "%1.15g", d);
cjson>       |                  ^
cjson> /build/source/cJSON.c:586:22: error: 'sprintf' was marked unused but was used [-Werror,-Wused-but-marked-unused]
cjson>   586 |             length = sprintf((char*)number_buffer, "%1.17g", d);
cjson>       |                      ^
cjson> /build/source/cJSON.c:1020:21: error: 'sprintf' was marked unused but was used [-Werror,-Wused-but-marked-unused]
cjson>  1020 |                     sprintf((char*)output_pointer, "u%04x", *input_pointer);
cjson>       |                     ^
cjson> 6 errors generated.
```

> In any case, the conditional needs to be tweaked to exclude the Darwin clang configuration (which works already)

Will do

--- a/pkgs/by-name/cj/cjson/package.nix
+++ b/pkgs/by-name/cj/cjson/package.nix
@@ -18,6 +18,10 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ cmake ];
 
+  cmakeFlags = lib.optional (stdenv.cc.isClang && !stdenv.hostPlatform.isDarwin) (
+    lib.cmakeBool "ENABLE_CUSTOM_COMPILER_FLAGS" false
+  );
+
   # cJSON actually uses C99 standard, not C89
   # https://github.com/DaveGamble/cJSON/issues/275
   postPatch = ''

